core/Db/Settings.php

```diff
@@ -32,9 +32,14 @@ public function getUsedCharset()
         return strtolower($this->getDbSetting('charset'));
     }
 
+    public function getRowFormat()
+    {
+       return $this->getDbSetting('disable_dynamic_row_format')?'':'ROW_FORMAT=DYNAMIC';
```

Comment:
The setting `disable_dynamic_row_format` doesn't exist in database configuration and would need to be added here: https://github.com/matomo-org/matomo/blob/7e0b0e602068a558a84f1e16fde5e7e6cb0b7fa4/config/global.ini.php#L43-L47

We can't simply use the dynamic row format for all instances, as some might break. So we need to disable that feature by default in `global.ini.php` and overwrite the value automatically if that feature is supported. This needs to be done while installing Matomo like here:
https://github.com/matomo-org/matomo/blob/09c012345d6913364af475598872105ff2fc854a/plugins/Installation/Controller.php#L608
and within a new updates script like this one https://github.com/matomo-org/matomo/blob/03410d62c6885010bc0968dca9ebe5e23b7c8e87/core/Updates/4.4.0-b1.php#L31-L36
To check if the feature is supported guess we could simply check if `DbHelper::getDefaultCharset()` returns `utf8mb4` or not.

Btw. Imho using some more spaces makes the code easier to read. e.g.
```php
return $this->getDbSetting('disable_dynamic_row_format') ? '' : 'ROW_FORMAT=DYNAMIC';
```

Reply:
thx sgiehL, updated it